ggml.c

```diff
@@ -5350,6 +5389,52 @@ GGML_API struct ggml_tensor * ggml_conv_transpose_1d(
     return result;
 }
 
+// ggml_conv_depthwise
+struct ggml_tensor * ggml_conv_depthwise_2d(
```

Comment:
Can the `ggml_conv_depthwise_2d` operator be represented as a combination of existing operators. For example, `ggml_conv_2d` is expressed via `ggml_im2col`, `ggml_mul_mat` and `ggml_reshape`:

https://github.com/ggerganov/llama.cpp/blob/ddb008d845cd50bb090bf051f570130524042936/ggml.c#L5404-L5429

If yes, it would be a better approach since it will allow directly GPU offload support

Reply:
Reviewing the code implementation in PyTorch, it seems that this operation (ggml_conv_depthwise_2d) is a conv2d.

Reply:
@ggerganov Thanks for your advices, We remove the `ggml_conv_depthwise_2d` native implementation, reconstruct it by `ggml_im2col`, `ggml_mul_mat`and `ggml_reshape`

Reply:
Nice! Does it run on the GPU now or is it still missing some operators?

Reply:
@ggerganov it can't run on GPU currently. there are two problems: 1. `ggml-cuda` doesn't support `pool2d` operator; 2. `ggml_cuda_can_mul_mat` requires `ne0>=32, ne1>=32, ne2>=3`, but `depthwise conv` with `kernel 3x3`'s `ne=[0]=9`. What is your opinions about the two problems ?

Reply:
We should extend support for these eventually